src/mscorlib/src/System/Reflection/AssemblyName.cs

```diff
@@ -533,7 +571,83 @@ internal string GetNameWithPublicKey()
         private extern byte[] nGetPublicKeyToken();
     
         [System.Security.SecurityCritical]  // auto-generated
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        static internal extern String EscapeCodeBase(String codeBase);
+        static internal String EscapeCodeBase(String codebase)
+        {
+            return EscapeString(codebase, false, true, true);
+        }
+
+        static internal String EscapeString(String str, bool escapeReserved, bool escapeHex, bool escapeBrackets)
```

Comment:
It looks like most of this functionality is already implemented in CoreFX UriHelper.EscapeString, which Uri.EscapeUriString uses. Although Uri.EscapeUriString seems to re-escape hex-escaped values, UriHelper.EscapeString can be configured as needed. Would it be better to call that through reflection instead, or to move that implementation into CoreLib?